### DISCUSSION TITLE
calling snap_manageState from within a cronjob hangs

### DISCUSSION THREAD

**Discussion by javaspeak** [MetaMask/snaps#2055] [Q&A]
I have code like this called from a snap cronjob:

```
     const snapStateRecord
      = await snap.request({
        method: 'snap_manageState',
        params: { operation: 'get' },
      });
```

I have a console log before and after the call and the log before the call prints.  The call after does not.  It is like the code is hanging?

Are we allowed to access snap storage from within a snap cronjob?

In the manifest I have this permission:

"snap_manageState": {},

**Comment by Montoya** [comment]
To access Snap storage from a cron job, you need to use the `encrypted` flag and set it to false. This will make it so you use unencrypted state, which is the only state that can be accessed from a cron job when MetaMask is locked. Encrypted state requires MetaMask to be unlocked. 

Example code for the flag is here: https://github.com/MetaMask/snaps/blob/main/packages/examples/packages/manage-state/src/index.ts#L37

Also, I am assuming you have the cron job endowment in your permissions? 
https://docs.metamask.io/snaps/reference/permissions/#endowmentcronjob

You have to specify the job (a function name that will be called) and you have to provide the timing as a string.
